libazureinit/src/health.rs

```diff
@@ -78,76 +130,140 @@ impl ReportableError {
             vm_id: vm_id.into(),
             timestamp: Utc::now(),
             pps_type: Some("None".to_string()),
-            extra: HashMap::new(),
+            supporting_data: HashMap::new(),
         }
     }
 
-    fn with_extra(mut self, key: &str, value: &str) -> Self {
-        self.extra.insert(key.to_string(), value.to_string());
+    /// Adds a key-value pair to the supporting data for the report.
+    pub fn with_supporting_data(mut self, key: &str, value: &str) -> Self {
+        self.supporting_data
+            .insert(key.to_string(), value.to_string());
         self
     }
+
+    /// Serializes the reportable error as a pipe-delimited KVP entry.
+    pub fn as_encoded_report(&self) -> String {
```

Comment:
this should take the parameters that may be unknown from the caller creating the reportable error.

e.g. vm_id, pps_type, etc.